### PR TITLE
SAM-3016 display student IP to instructor

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

**Comment by joaquinmarques** [comment]
cla has been sent

**Comment by ern** [comment]
Minor change remove the comments as they just get in the way and are uninformative.

Since this is going in a log does VARCHAR(99) seem appropriate for storing an IP Address? which could be stored as an int or a long... i.e. ipv4 = 4 bytes vs 45 bytes * the number of rows.

However I can understand wanting for this to be in dotted, though there exists SQL functions to parse into the correct format.

**Comment by buckett** [comment]
I'm pretty sure we want to cope with IPv4 and IPv6.

**Comment by joaquinmarques** [comment]
Comments from source have been deleted

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java [hunk 1]
```diff
@@ -442,7 +442,10 @@ public void processAction(ActionEvent ae) throws
                   eventLogData.setErrorMsg(eventLogMessages.getString("no_submission"));
                   eventLogData.setEndDate(null);
                   eventLogData.setEclipseTime(null);
-                      
+                  				  
+				  String thisIp = ( (javax.servlet.http.HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest()).getRemoteAddr();
+				  eventLogData.setIpAddress(thisIp);
+
```
Comment (ottenhoff) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java]: Can you fix the tabs here?

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java [hunk 2]
```diff
@@ -572,7 +575,10 @@ else if (action == DeliveryBean.TAKE_ASSESSMENT_VIA_URL) {
     		eventLogData.setEndDate(null);
     		eventLogData.setEclipseTime(null);
     	}
-    	
+    			
+		String thisIp = ( (javax.servlet.http.HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest()).getRemoteAddr();
+		eventLogData.setIpAddress(thisIp);
+
```
Comment (ottenhoff) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java]: Can you fix tabs/spaces here?

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/LinearAccessDeliveryActionListener.java [hunk 3]
```diff
@@ -197,7 +197,10 @@ public void processAction(ActionEvent ae) throws
           eventLogData.setErrorMsg(eventLogMessages.getString("no_submission"));
           eventLogData.setEndDate(null);
           eventLogData.setEclipseTime(null);
-              
+          		  
+		  String thisIp = ( (javax.servlet.http.HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest()).getRemoteAddr();
+		  eventLogData.setIpAddress(thisIp);
+
```
Comment (ottenhoff) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/LinearAccessDeliveryActionListener.java]: tabs/spaces

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/queue/delivery/SubmitTimedAssessmentThread.java [hunk 4]
```diff
@@ -119,6 +122,10 @@ public void run(){
         	  eventLogData.setEclipseTime(null); 
         	  eventLogData.setErrorMsg(eventLogMessages.getString("error_take"));
           }
+		  			
+			String thisIp = ( (javax.servlet.http.HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest()).getRemoteAddr();
+			eventLogData.setIpAddress(thisIp);
+
```
Comment (ottenhoff) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/queue/delivery/SubmitTimedAssessmentThread.java]: tabs/spaces

File: samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/PublishedAssessment.hbm.xml [hunk 5]
```diff
@@ -239,5 +239,6 @@
     <property name="endDate" type="timestamp" column="ENDDATE" not-null="false" />
     <property name="eclipseTime" type="integer" column="ECLIPSETIME" not-null="false" />
     <property name="errorMsg" type="string" length="255" column="ERRORMSG" not-null="false" />
+	<property name="ipAddress" type="string" length="99" column="IPADDRESS" not-null="false" />
```
Comment (ottenhoff) [inline samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/PublishedAssessment.hbm.xml]: tabs/spaces

**Comment by joaquinmarques** [comment]
Fix tabs and spaces

**Pull request merged**